lua/telescope/pickers.lua

```diff
@@ -512,6 +520,14 @@ function Picker:find()
 
         status_updater { completed = false }
         self._on_lines(...)
+
+        if self.sorting_strategy then
+          if self.sorting_strategy == "ascending" then
+            self:move_selection(1)
+          else
+            self:move_selection(-1)
```

Comment:
I can't tell what this block is actually for. Is it just another way to force a redraw?

Reply:
Yeah. It's a way to force redraw the results win. The reason why not use `nvim_win_set_cursor` is in this case, the cursor's position should be recalculated.

Reply:
I'm not sure what this fixes, but I didn't need it to fix the issues I am seeing.

Reply:
just use another `nvim_win_call` with `redraw` and the other window?

Reply:
Yeah. This works for me:

```lua
        vim.api.nvim_win_call(self.results_win, vim.cmd.redraw)
```

Reply:
Scratch that, that doesn't always work, but this seems to:

```lua
        self:move_selection(self.sorting_strategy == "ascending" and 1 or -1)
```

Reply:
Try these steps to reproduce the issue:

- type a pattern to filter the results, and the number of results should be less than the height of results win
- delete char, and the number of results now is still less than the height of results win

In this case, the results win is empty.

I try `vim.api.nvim_win_call(self.results_win, function() vim.cmd "redraw!" end) ` to force redraw the results win. It works. But in this case, my statusline disappeared.